sdk/nodejs/runtime/rpc.ts

```diff
@@ -419,6 +419,19 @@ export async function serializeProperty(ctx: string, prop: Input<any>, dependent
         return result;
     }
 
+    if (prop instanceof Map) {
```

Comment:
Our SDK code generator for Node.js emits map types as `{[key: string]: T}` rather than `Map<string, T>`, so it's unclear to me if support for serializing `Map`s is necessary.

https://github.com/pulumi/pulumi/blob/67bcd73a15cf718e1ac2dccddb6d5d5bb3d9504d/pkg/codegen/nodejs/gen.go#L216-L217